channeldb/migrations.go

```diff
@@ -514,19 +514,41 @@ func migratePruneEdgeUpdateIndex(tx *bolt.Tx) error {
 			err)
 	}
 
-	// With the existing edge policies gathered, we'll recreate the index
-	// and populate it with the correct entries.
-	//
-	// NOTE: In bolt DB, calling Delete() on an iterator will actually move
-	// it to the NEXT item. As a result, we call First() here again to go
-	// back to the front after the item has been deleted.
-	updateCursor := edgeUpdateIndex.Cursor()
-	for k, _ := updateCursor.First(); k != nil; k, _ = updateCursor.First() {
-		if err := updateCursor.Delete(); err != nil {
+	log.Info("Constructing set of edge update entries to purge.")
+
+	// Build the set of keys that we will remove from the edge update index.
+	var updateKeysToRemove [][]byte
+	err = edgeUpdateIndex.ForEach(func(updKey, _ []byte) error {
+		// All valid entries in the edge update index are 16 bytes,
+		// consisting of an 8-byte timestamp and an 8-byte short channel
+		// id. We'll skip any entries not matching this format.
+		if len(updKey) != 16 {
+			return nil
```

Comment:
if these keys are invalid, shouldn't we delete them?

Reply:
Fixed